### DISCUSSION TITLE
Support "async" validation functions

### DISCUSSION THREAD

**Discussion by barrywhart** [zauberzeug/nicegui#4004] [Ideas / Feature Requests]
Currently, it appears that asynchronous validation functions are not supported, resulting in an error, "TypeError: Type is not JSON serializable: coroutine". This would be a useful feature. I have a case where an input (text) field needs to be validated for uniqueness, and this requires querying the application database, which is an `async` call.

**Comment by falkoschindler** [comment]
Hi @barrywhart,

Adding support for async validation functions might be a pretty handy feature. It's a bit tricky to fit into the existing API, but I just created a draft PR #4024.

**Reply by falkoschindler** [reply]
This feature has been released in version 2.7.0. ✔️

**Comment by barrywhart** [comment]
Great, thanks for the quick turnaround! I'll update my existing code to use this. I was using a hacky workaround before.
